### ISSUE TITLE
Add support for utf-8-sig encoded .md5 files

### ISSUE THREAD

**Issue by sellth** [bihealth/sodar-server#1818]
### Describe the Bug
Windows programs such as Notepad or [Powershell](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_character_encoding?view=powershell-7.3#the-byte-order-mark) attach byte-order marks (BOM) to UTF-8 encoded text files. These files look and behave the same as "normal" UTF-8 encoded files, but SODAR fails to decode them, probably [here](https://github.com/bihealth/sodar-server/blob/455c32ccab5186791feebd14abe023749c17b193/taskflowbackend/tasks/irods_tasks.py#L566C57-L566C63). This causes LZ validation to fail every time with a `(File: ; iRODS: xyz)` message.

This potentially blocks the use of native Windows tools such as Powershell scripts in SODAR upload workflows. As Windows is the standard Desktop of Charité and many manufacturers also ship Windows PCs with their instruments, I think this issue has some merit.

### How to Reproduce

1. Compute a .md5 file using the Windows powershell or [manually add a BOM](https://stackoverflow.com/questions/3127436/adding-bom-to-utf-8-files) to an existing .md5 file. 
2. Upload the corresponding data file + .md5 into a LZ. 
3. Validate LZ.

### Expected Behavior

- SODAR should handle/remove BOMs in .md5 files.
- LZ should validate.

### Workaround
Encode .md5 files with ANSI instead. Only hash sum part is used by SODAR and should never contain non-ANSI characters.

**Comment by mikkonie** [comment]
Technically this is not a bug, as support for such files was never a requirement ;)

This is probably trivial to implement, I'll take a look into it.

**Comment by mikkonie** [comment]
On further thinking, the question becomes: **should** we support these files?

The .md5 files are stored in the sample data repository so they can also be verified when downloading data from iRODS. This means the users will also need to support this file format. Do the tools supported by (most) users always support these? If not, we should instead require standard formats.I believe most of our users don't use Powershell.

**Comment by sellth** [comment]
Can SODAR just strip the BOM from these files i.e. convert them to regular utf8 during LZ move? 

Windows compatibility is important, it lowers the barrier of entry for many users. Plus, this incompatibility is not easily apparent and thus can lead to great frustration.

**Comment by mikkonie** [comment]
Replacing the file in iRODS is possible of course, but again this is a question on whether it's worth the effort. You say "many users", but how many Powershell users are we expecting to see? This is the first time I've heard it mentioned in the context of SODAR in 7 years..

**Comment by sellth** [comment]
That's true, but CUBI never supplied Windows users with SODAR tools in the first place. This is about enabling the usage of Windows-native tools in the first place. iRODS upload via Cyberduck + md5 computation via Powershell script is a viable workflow I'm trying to establish, but it's soft-blocked by this utf-8-sig thing.

**Comment by mikkonie** [comment]
I'll raise this in our next tech meeting to get some second and third opinions on what would be the best approach. Meanwhile, as a mental note, [here are some examples of handling the BOM headers in Python](https://stackoverflow.com/questions/13590749/reading-unicode-file-data-with-bom-chars-in-python).

**Comment by mziehm** [comment]
I ran into the same issue trying to create way for our lab member to upload data to SODAR (with the least amount of extra tools to install and maintain) and was pointed to this issue by @sellth.
While I understand the considerations discussed above. Please at least, provide clear instructions on the md5 file requirements (see also #2032).

**Comment by sellth** [comment]
I ran into this again when trying to teach the Metabolomics guys to upload files to iRODS. As there apparently is no way to write normal utf-8 encoded text files with the Windows Powershell [without changing system wide locale settings](https://stackoverflow.com/a/65192064) (!!!) it would be great if SODAR could also read utf-8-sig encoded md5 files.

**Comment by mikkonie** [comment]
I'll take a look at this next week. If feasible, I'll still sneak it in v1.0.0 which is currently blocked waiting for production iRODS server setup anyway.

**Comment by mikkonie** [comment]
Done. Please note that this only works with UTF-8 BOM headers, other encodings are not supported. If some other exotic cases arise, please open new issue(s) accordingly.
